File: 1138-grumpy-bookstore-owner/1138-grumpy-bookstore-owner.py
class Solution:
    def maxSatisfied(self, customers: List[int], grumpy: List[int], minutes: int) -> int:
        defSatis = 0
        for i in range(len(customers)):
            if(grumpy[i] == 0):
                defSatis += customers[i]
        
        currSum = defSatis
        for i in range(minutes):
            if(grumpy[i] == 1):
                currSum += customers[i]
        maxSum = currSum
        for i in range(minutes, len(customers)):
            if(grumpy[i - minutes] == 1):
                currSum -= customers[i - minutes]
            if(grumpy[i] == 1):
                currSum += customers[i]

            maxSum = max(currSum, maxSum)
        return maxSum

File: 1138-grumpy-bookstore-owner/1138-grumpy-bookstore-owner.java
class Solution {
    public int maxSatisfied(int[] customers, int[] grumpy, int minutes) {
        int defaultSatisfied = 0;
        for(int i = 0; i < customers.length; i++)
        {
            if(grumpy[i] == 0)
            {
                defaultSatisfied += customers[i];
            }
        }

        int max = defaultSatisfied;
        for(int i = 0; i < customers.length; i++)
        {
            int sum = defaultSatisfied;
            for(int m = i; m < minutes + i && m < customers.length; m++)
            {
                if(grumpy[m] == 1) //if he was grumpy at this time i can flip to non grumpy
                {
                    sum += customers[m];
                }
            }
            max = Math.max(max, sum);
        }
        return max;
    }
}